### PR TITLE
Adds self-supervision and fine-tuning on AQuA rationales

### PR DESCRIPTION
**Pull request by piotrekp1**
Hello! 
This commit does few things:
- Creates pipeline for training on AQuA
- enables loading bert models pretrained in trax and fine-tuning them
- adds metrics for pretraining when using both losses (ClsAccuracy , MLMAccuracy)

### CODE REVIEW

File: trax/models/research/bert.py [hunk 1]
```diff
@@ -255,6 +269,13 @@ def init_weights_and_state(self, input_signature):
     if self.init_checkpoint is None:
       return
 
+    if self.init_checkpoint.endswith('.pkl.gz'):
+      # file is a model saved in trax
+      model2 = BERT(head=functools.partial(BERTPretrainingHead, n_classes=2))
+      model2.init_from_file(self.init_checkpoint)
```
Comment (henrykmichalewski) [inline trax/models/research/bert.py:275]: Initializing models should be done using the default interface.

File: trax/data/tf_inputs.py [hunk 2]
```diff
@@ -1264,6 +1265,61 @@ def split_stream(generator=None):
...
+    ds = [json.loads(line) for line in f.readlines()]
+
+  def iterator(generator=None):
+    while True:
+      for el in ds:
+        yield el
+  return iterator
+
+
+def AQuAExtendedQuestion():
```
Comment (henrykmichalewski) [inline trax/data/tf_inputs.py:1281]: Can you add docstrings as in https://google.github.io/styleguide/pyguide.html or as visible in other places in TRAX? Also, we will need a short info how to work with the new loss and the new dataset.

**Review by henrykmichalewski — COMMENTED** [review]
Please, describe how do you imagine the self-supervised training. Is that supposed to be done only using the Aqua dataset? Can you link some preliminary training results (tensorboard.dev) in particular for lighter tasks such as fine-tuning on Aqua?

**Comment by piotrekp1** [comment]
for self-supervision model should be loaded with trained bert-base-uncased weights and then trained on pretext task - ROP and MLM on AQuA-RAT but with rationales. After that model is fine-tuned for predicting correct answer and in this stage it doesn't utilize rationales and predicts correct answer directly from the question

I ran few epochs of fine-tuning bert to AQuA (without self-supevision), here are the results:
https://tensorboard.dev/experiment/hmyWkdEbTvyYgDk3WA2jeA/#scalars&_smoothingWeight=0
(0/eval is validation set and 1/eval is test set).

File: trax/models/research/bert.py (hunk 1 above)
Comment (piotrekp1) [reply trax/models/research/bert.py:275]: I'm not sure if this is possible with default interface, the problem is that loaded model has bert core and pretraining head and we're interested only in loading the core, not the head. If it is possible could you give me a hint or direct me to some snippet that does something similar?
